W-18940999-merge-MS2025-06-ACB-to-latest-mk (#474)
mulesoft/docs-code-builder: modules/ROOT/pages/api-ai-create-spec.adoc

The Generative API Specification feature unlocks AI capabilities for API design journeys by generating and mocking API specs from natural language. The feature reduces the time spent on API design by simplifying the creation of rigid, syntax-heavy specs. You can access the feature after creating an API project that's in the supported RAML 1.0 or OAS 3.0 format or from the Open Agentforce icon within the project. The system supports multi-turn conversations and provides validated API specifications.

[[before-you-begin]]
== Before You Begin

Before you start creating your API spec:

* xref:start-acb.adoc[Set up and access the web or desktop IDE].
* Make sure you have the xref:start-configure-permissions.adoc#permissions[required Anypoint Code Builder permissions].
* Make sure you have these permissions to use the Anypoint Code Builder Agentforce feature:
** Anypoint Code Builder Developer
** Mule Developer Generative AI User
* Make sure Einstein is enabled in Access Management. For more information, see xref:access-management::enabling-einstein.adoc[Enabling Einstein for Anypoint Platform].
* Log in to Anypoint Code Builder with your Anypoint Platform credentials.

> * xref:des-create-api-specs.adoc[Create an API spec project] in Anypoint Code Builder in the supported RAML 1.0 or OAS 3.0 format, and then select a business group. The business group must have the xref:access-agentforce.adoc#agentforce[Agentforce feature enabled]. If you don't have the Agentforce feature after selecting a business group, contact your Anypoint Platform administrator to enable the feature.

Reviewer: Should Agentforce be changed to generative AI here?

Author: ```suggestion
* xref:des-create-api-specs.adoc[Create an API spec project] in Anypoint Code Builder in the supported RAML 1.0 or OAS 3.0 format, and then select a business group. The business group must have xref:access-agentforce.adoc#agentforce[Einstein generative AI enabled]. If you don't have Einstein generative AI after selecting a business group, contact your Anypoint Platform administrator to enable the feature.
```

* Object
* String
* Time only

Optionally, provide a description, format, minimum and maximum length, default value, and example for the header.

*Query Parameter Requirements*

Query parameters must include the name, whether it's required, and one of the following types:

* Array
* Boolean
* Date only
* DateTime
* DateTime only
* File
* nil
* Number
* Object
* String
* Time only 

Optionally, provide a description, format, minimum and maximum length, default value, and example for the query parameter.
====

. Send your message. The agent creates a validated API spec.

[NOTE]
Governance isn't included in the validation.

[start=3]
. Review the spec and continue the conversation if you'd like to make changes.
. Click *Replace* to add the generated API spec to your project. All file content is replaced by the inserted code.
+
You can also copy the generated API spec to your clipboard by clicking *Copy*.

== Give Feedback on Agent Responses

Share your feedback about the agent's response by clicking thumbs up or thumbs down.

== View Conversation History

To view your past conversations, click *Open chat history*. Your last 30 days of conversations show.

== Start a New Chat

> You can only create a new API spec if you have a project open and that's in the supported RAML 1.0 or OAS 3.0 format. Click *Create API Spec* from the *New Chat* dropdown to create a new API spec.

Reviewer: ```suggestion
You can only create an API spec if you have a project open in the supported RAML 1.0 or OAS 3.0 format. Click *Create API Spec* from the *New Chat* dropdown to create a new API spec.
```
